src/main/java/org/archive/url/BasicURLCanonicalizer.java

```diff
@@ -74,6 +75,36 @@ public void canonicalize(HandyURL url) {
 		url.setPath(escapeOnce(normalizePath(path)));
 	}
 
+	/**
+	 * Normalize dots in the host name.
+	 *
+	 * @param host
+	 * @return host name with all sequences of dots replaced with a single dot,
+	 *         and all leading and trailing dots removed
+	 */
+	private String normalizeDots(String host) {
+		if (host.indexOf('.') == -1) {
+			return host;
+		}
+		int start = 0, end = host.length();
+		boolean changed = false;
+		while (host.charAt(start) == '.') {
```

Comment:
Prevent StringIndexOutOfBoundsException for input like "." or "....":

```suggestion
		while (start < end && host.charAt(start) == '.') {
```

Reply:
Thanks, good catch!